integrations/oci/metrics/metrics/src/main/java/io/helidon/integrations/oci/metrics/OciMetricsSupport.java

```diff
@@ -0,0 +1,527 @@
+
+        if (allMetricDataDetails.size() > 0) {
+            while (true) {
+                if (allMetricDataDetails.size() > batchSize) {
+                    postBatch(allMetricDataDetails.subList(0, batchSize));
+                    // discard metrics that had been posted
+                    allMetricDataDetails.subList(0, batchSize).clear();
+                    if (batchDelay > 0L) {
+                        try {
+                            schedulingTimeUnit.sleep(batchDelay);
+                        } catch (InterruptedException ignore) {
+                        }
+                    }
+                } else {
+                    postBatch(allMetricDataDetails);
+                    break;
+                }
+            }
+        }
+    }
+
+    private void postBatch(List<MetricDataDetails> metricDataDetailsList) {
+        PostMetricDataDetails postMetricDataDetails = PostMetricDataDetails.builder()
+                .metricData(metricDataDetailsList)
+                .build();
+
+        PostMetricDataRequest postMetricDataRequest = PostMetricDataRequest.builder()
+                .postMetricDataDetails(postMetricDataDetails)
+                .build();
+
+        LOGGER.finest(String.format("Pushing %d metrics to OCI", metricDataDetailsList.size()));
+        if (LOGGER.isLoggable(Level.FINEST)) {
+            metricDataDetailsList
+                    .stream()
+                    .forEach(m -> {
+                        LOGGER.finest(String.format(
+                                "Metric details: name=%s, namespace=%s, dimensions=%s, "
+                                        + "datapoints.timestamp=%s, datapoints.value=%f, metadata=%s",
+                                m.getName(),
+                                m.getNamespace(),
+                                m.getDimensions(),
+                                m.getDatapoints().get(0).getTimestamp(),
+                                m.getDatapoints().get(0).getValue(),
+                                m.getMetadata()));
+                    });
+        }
+        String originalMonitoringEndpoint = this.monitoringClient.getEndpoint();
+        try {
+            // Use the ingestion endpoint for posting
+            this.monitoringClient.setEndpoint(
+                    monitoringClient.getEndpoint().replaceFirst("telemetry\\.", "telemetry-ingestion."));
+            this.monitoringClient.postMetricData(postMetricDataRequest);
+            LOGGER.finest(String.format("Successfully posted %d metrics to OCI", metricDataDetailsList.size()));
+        } catch (Throwable e) {
+            LOGGER.warning(String.format("Unable to send metrics to OCI: %s", e.getMessage()));
+        } finally {
+            // restore original endpoint
+            this.monitoringClient.setEndpoint(originalMonitoringEndpoint);
+        }
+    }
+
+    @Override
+    public void update(Routing.Rules rules) {
+        rules.onNewWebServer(this::prepareShutdown);
```

Comment:
Does it make sense to move this to after line 255? If we don't need the executor (because the feature is disabled or there are no metrics to report) then we won't need the shutdown processing.

Reply:
implementing